.github/workflows/test.yml

```diff
@@ -0,0 +1,47 @@
+name: Run tests and lint
+
+on:
+  push:
+    branches: ['main']
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ['3.11', '3.12']
```

Comment:
Are the the lowest (and highest) supported versions? If not it might help to run Python against the lowest supported version to make sure devs to slip up something like `f"{foo=}"`, the walrus or whatever new joy they'd like to use.